bench/blockstore.js

```diff
@@ -0,0 +1,476 @@
+/*!
+ * bench/blockstore.js - benchmark blockstore for bcoin
+ *
+ * This can be run to benchmark the performance of the blockstore
+ * module for writing, reading and pruning block data. Results are
+ * written to stdout as JSON or formated bench results.
+ *
+ * Usage:
+ * node ./blockstore.js [--maxfile=<bytes>] [--total=<bytes>]
+ *                      [--location=<path>] [--store=<name>]
+ *                      [--output=<name>] [--unsafe]
+ *
+ * Options:
+ * - `maxfile`  The maximum file size (applies to "file" store).
+ * - `total`    The total number of block bytes to write.
+ * - `location` The location to store block data.
+ * - `store`    This can be "file" or "level".
+ * - `output`   This can be "json", "bench" or "benchjson".
+ * - `unsafe`   This will allocate block data directly from memory
+ *              instead of random, it is faster.
+ *
+ * Copyright (c) 2019, Braydon Fuller (MIT License).
+ * https://github.com/bcoin-org/bcoin
+ */
+
+'use strict';
+
+process.title = 'blockstore-bench';
+
+const {isAbsolute} = require('path');
+const {mkdirp} = require('bfile');
+const random = require('bcrypto/lib/random');
+const {BufferMap} = require('buffer-map');
+
+const {
+  FileBlockStore,
+  LevelBlockStore
+} = require('../lib/blockstore');
+
+const config = {
+  'maxfile': {
+    value: true,
+    parse: a => parseInt(a),
+    valid: a => Number.isSafeInteger(a),
+    fallback: 128 * 1024 * 1024
+  },
+  'total': {
+    value: true,
+    parse: a => parseInt(a),
+    valid: a => Number.isSafeInteger(a),
+    fallback: 3 * 1024 * 1024 * 1024
+  },
+  'location': {
+    value: true,
+    valid: a => isAbsolute(a),
+    fallback: '/tmp/bcoin-bench-blockstore'
```

Comment:
This is not OS agnostic and should be able to run on Windows

Reply:
This may be out-of-scope, the documentation mentions that bcoin has not been tested on windows, see https://github.com/bcoin-org/bcoin/blob/master/docs/getting-started.md#requirements

Reply:
This location can be configured here for the benchmark, this is a fallback value and can be change via the `--location` argument, as is a common case for testing various disks.

I've added a commit to make the test directory configurable for various environments, including GNU/Linux and BSD variations, and users setups, that may use an alternative `TEMP` path. In addition, the test directory is made unique so that parallel runs will not conflict.